theory-test-admission-candidate-app/app/server/controllers/candidateDetailsController.js

```diff
@@ -10,15 +11,18 @@ export default class CandidateDetailsController {
 	checkBooking(req, res) {
 		const admissionsId = uuid();
 		const candidateDLN = req.body.DLN.toString().toUpperCase();
-		this.candidateBookingService.admissionsId = admissionsId;
-		this.candidateBookingService.candidateDLN = candidateDLN;
-		const hasBooking = this.candidateBookingService.checkCandidateBooking();
-		sessionStorage(req, res, 'bookingResult', { hasBooking, admissionsId });
-		if (hasBooking) {
-			sessionStorage(req, res, 'videoTermsAgreed', false);
-			res.redirect('/candidate/prepare-video');
-		} else {
-			res.redirect('/candidate/report-reception');
-		}
+		CandidateCheckBookingService.checkCandidateBooking(admissionsId, candidateDLN, (hasBooking) => {
```

Comment:
This abstraction is not correct.
If we are in the user interface, we are not trying to *check* the candidate's booking.
If anything, we are trying to retrieve it.

Reply:
done